### ISSUE TITLE
On OS X, all functionality using command-click is not usable

### ISSUE THREAD

**Issue by poke1024** [godotengine/godot#10761]
On OS X all command-clicks are mapped to a right button click (which is the usual convention under OS X); see `platform/osx/os_osx.mm`:

```
- (void)mouseDown:(NSEvent *)event {
	if (([event modifierFlags] & NSControlKeyMask)) {
		mouse_down_control = true;
		_mouseDownEvent(event, BUTTON_RIGHT, BUTTON_MASK_RIGHT, true);
	} else {
		mouse_down_control = false;
		_mouseDownEvent(event, BUTTON_LEFT, BUTTON_MASK_LEFT, true);
	}
}
```

This means it's not possible to generate a left button click with a control key (since it's immediately mapped to a right click). Any code relying on such command-left click (e.g. adding points with the current polygon 2d code) cannot be reached under OS X thus.

I'm not sure what a good fix would be. One option might be to introduce a new combination for OS X like ctrl-alt-cmd-left click, that is usually used nowhere in Godot, and that is mapped to the usual control-click.

**Comment by toger5** [comment]
I also ran into an issue caused by this.

I tried to implement ctrl click drag on spin sliders. (which does not work on OS X because the event never arrives as an left click + ctrl event...)

my favourite solution is to just let OS X users suffer from godot not following this os convention ;)

I'm honestly not sure how many ppl use it since the two finger click on trackpads became really common...

I think your other proposal is really smart although I don't like it as a user...

I would prefer to be forced to use two finger click for right click.

maybe we should remove the "map ctrl to right click" behaviour on OS X but make give it an option for ppl who really want it and are okay with ctrl-alt-cmd-left click for control click.

@marcelofg55 what are your thoughts since you implemented the ctrl -> right click?

**Comment by marcelofg55** [comment]
@toger5 I couldn't think a good solution for this myself but #19092 sounds like a good option to fix this

**Comment by reduz** [comment]
I don't really understand this issue. In most Godot editors, "control" should not ever be used, simply use command-click, which will map to the Apple shortcut for this on OSX and Control on Windows / Linux Desktop. This way, control is free to work as control-click on OSX.

**Comment by akien-mga** [comment]
> On OS X all command-clicks are mapped to a right button click (which is the usual convention under OS X)

This issue mentions *Cmd*+Click, but the code reads *Control* and that's also the standard analogy for Right Click on macOS. Can we get a confirmation that *Cmd*+Click does NOT generate a Right Click event currently? If so, what @reduz mentioned is the proper fix. All shortcuts should use *Cmd* which is translated to *Ctrl* on non-macOS.

**Comment by marcelofg55** [comment]
What @reduz said sounds like the proper fix I think.

**Comment by k7771911** [comment]
With the latest nightly build some of the shortcuts started to work. Bit of a problem with the Editor Next tab shortcut Command-Tab which is a macOS system shortcut (next application, equivalent of the windows alt-tab) Trying to rebind to Control-Tab does not work, somehow the Tab key does not register when Control is pressed

**Comment by k7771911** [comment]
The Pick Tile menu entry still says the shortcut is Control but Command-Click works fine

**Comment by k7771911** [comment]
Would it be possible to backport the Mac shortcut fix into the 3.0 line? Would make the editor so much more usable on Mac.

**Comment by akien-mga** [comment]
@k7771911 It's marked for cherry-picking to the 3.0 branch since June, but all such changes are reviewed and tested by @hpvb before eventually making it to the stable branch.
